src/vscode-dts/vscode.proposed.treeViewRenderLabelIcons.d.ts

```diff
@@ -0,0 +1,18 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// @kycutler https://github.com/microsoft/vscode/issues/271523
+
+	export interface TreeViewOptions<T> {
+		/**
+		 * Whether to render icons in tree item labels. When `true`, icons embedded in the label
+		 * (e.g., codicons like `$(icon-name)`) will be rendered as icons. When `false`, they will
+		 * be rendered as plain text. Defaults to `false`.
+		 */
+		renderLabelIcons?: boolean;
```

Comment:
Instead of having an option here, I think we should allow `TreeItemLabel`'s `label`, to be `string | MarkdownString`.

Reply:
`TreeView` uses `IResourceLabel`s for the item labels, which end up being far more complex than just simple strings. But they do support icon rendering. Rather than trying to add Markdown support to those, or updating `TreeView` to not use resource labels, I figure the easiest way is to just use that flag.

The reason it's a flag on the tree view and not in the label itself is due to the way tree views are rendered -- from my understanding the tree items are rendered using templates, and icon support is configured in the template. So it's not really feasible to configure on each individual item.